SolveSATWithGrover/Tasks.qs

```diff
@@ -97,54 +94,44 @@ namespace Quantum.Kata.GroversAlgorithm {
-    // For 2-SAT problems, f(x) is represented as a conjunction (an AND operation) of M clauses on N variables, 
-    // and each clause is a disjunction (an OR operation) of two variables or negated variables:
-    //      f(x) = ∧ᵢ (yᵢ₀ ∨ yᵢ₁), yᵢₖ = either xⱼ or ¬xⱼ for some j in {0, ..., N-1}
-    //
-    // For example, XOR function f(x) = x₀ ⊕ x₁ can be represented as a 2-SAT problem 
-    // using N = 2 variables and M = 2 clauses as follows:
-    //      f(x) = x₀ ⊕ x₁ = (x₀ ∨ x₁) ∧ (¬x₀ ∨ ¬x₁)
-    //
+    // Task 1.5. Evaluate one clause of a SAT formula
+    // 
+    // For k-SAT problems, f(x) is represented as a conjunction (an AND operation) of several clauses on N variables, 
+    // and each clause is a disjunction (an OR operation) of one or several variables or negated variables:
+    //      clause(x) = ∨ₖ yᵢₖ, where yᵢₖ = either xⱼ or ¬xⱼ for some j in {0, ..., N-1}
+    // 
+    // For example, one of the possible clauses on two variables is 
+    //      clause(x) = x₀ ∨ ¬x₁
+    // 
     // Inputs:
     //      1) N qubits in an arbitrary state |x⟩ (input/query register)
     //      2) a qubit in an arbitrary state |y⟩ (target qubit)
-    //      3) a 2-dimensional array of tuples "problem" which describes the 2-SAT problem instance f(x).
+    //      3) a 1-dimensional array of tuples "clause" which describes one clause of a SAT problem instance clause(x).
     //
-    // i-th element of "problem" describes the i-th clause of f(x);
-    // it is an array of 2 tuples, each of them describing one component of the clause.
+    // "clause" is an array of one or more tuples, each of them describing one component of the clause.
     // Each tuple is an (Int, Bool) pair:
-    //  - the first element is the index of the variable xⱼ,
+    //  - the first element is the index j of the variable xⱼ,
     //  - the second element is true if the variable is included as itself (xⱼ) and false if it is included as a negation (¬xⱼ)
     // 
-    // Example 1:
-    // The oracle from the task 4.2 (OR oracle x₀ ∨ x₁) can be represented as [[(0, true), (1, true)]].
-    //
-    // Example 2:
-    // The oracle from the task 4.3 (XOR oracle x₀ ⊕ x₁) can be represented as [[(0, true), (1, true)], [(0, false), (1, false)]]
-    //
-    // Goal: Transform state |x, y⟩ into state |x, y ⊕ f(x)⟩ (⊕ is addition modulo 2).
-    //       Leave the query register in the same state it started in.
-    operation Oracle_2SAT (queryRegister : Qubit[], 
-                           target : Qubit, 
-                           problem : (Int, Bool)[][]) : Unit
-        is Adj {        
-            // ...
+    // Example:
+    // The clause clause(x) = x₀ ∨ ¬x₁ can be represented as [(0, true), (1, false)].
+    operation Oracle_SATClause (queryRegister : Qubit[], 
+                                target : Qubit, 
+                                clause : (Int, Bool)[]) : Unit is Adj {        
+        // ...
     }
 
 
     // Task 1.6. k-SAT problem oracle
```

Comment:
"k-SAT" implies that each clause has at most k variables. What you're building here is really a general SAT problem oracle, allowing an arbitrary number of variables in each clause.
```suggestion
    // Task 1.6. General SAT problem oracle
```